src/components/QuestionSubmissionForm.tsx

```diff
@@ -11,30 +12,29 @@ interface QuestionSubmissionFormInputs {
   stay_anonymous: boolean;
 }
 
-export function QuestionSubmissionForm({ session }: PageProps) {
+export function QuestionSubmissionForm() {
   // reference: https://react-hook-form.com/get-started#Quickstart
   const { register, handleSubmit, reset, formState } = useForm<QuestionSubmissionFormInputs>();
+  const { user } = useUser();
 
   const onSubmit: SubmitHandler<FieldValues> = async data => {
     console.log(data);
 
-    if (session === undefined || session === null) {
+    if (!user) {
       console.log('user not logged in.');
 
       return;
     }
 
-    let result;
-    try {
-      result = await fetch('/api/submit_question', {
-        headers: {
-          Authentication: session.access_token,
-        },
-        method: 'POST',
-        body: JSON.stringify(data),
-      });
-    } catch (err) {
-      console.log(err);
+    const asked_date = new Date(new Date().setDate(new Date().getDate() - Number(data['recency_weeks']) * 7));
+
+    data['asked_date'] = asked_date;
+    delete data['recency_weeks'];
+
+    const { data: result, error } = await supabaseClient.from('questions').insert([data]);
```

Comment:
If we make the insertion request from here and allow it, wouldn't people be able to insert any arbitrary data from here? E.g., set the is_approved to true?

Reply:
Only if you didn't set any policy for the ``questions`` table. If you added an RLS for ``questions``, only for authenticated people for example, so it would not be a problem.

Reply:
With RLS, logged-in people would still be able to set is_approved to true.

Most people might not do it, but that's a concern. Any thoughts about that?

Reply:
You can define a RLS for specific fields, like ``is_approved``, and define it like "is user is authenticated and has permission X, then ok, otherwise no".

Reply:
How can we do that?

Reply:
Following what Gary said on Discord:

![image](https://user-images.githubusercontent.com/25107942/194635844-e4d2d2ec-3fc8-4f26-8f73-e6118668f6bb.png)

It's a lot of Postgres stuff

Reply:
@Olyno first of all, who's Gary, and which Discord is this?

Secondly - all this seems to do is update the created_at field. How would it apply to this particular problem/application?

Reply:
The quoted code is of course an example given by Gary, a moderator and very active member of the official Supabase discord server. You just have to take what he did, and apply what he proposed to the case of this project.

Reply:
Cool. Did you test it to see if it works for our particular use case?

Reply:
I didn't test it on my side

Reply:
How do you know it works, then?